raphtory/src/algorithms/components/lcc.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{
+    algorithms::{components::connected_components::weakly_connected_components},
+    db::{
+        api::view::{StaticGraphViewOps, GraphViewOps},
+        graph::views::node_subgraph::NodeSubgraph,
+    },
+    prelude::Graph,
+};
+use std::collections::HashMap;
+
+/// Gives the large connected component of a graph.
+/// The large connected component is the largest (i.e., with the highest number of nodes)
+/// connected sub-graph of the network.
+///
+/// # Example Usage:
+///
+/// g.largest_connected_component()
+///
+/// # Returns:
+///
+/// A raphtory graph, which essentially is a sub-graph of the graph `g`
+///
+pub trait LargestConnectedComponent {
+    fn largest_connected_component(&self) -> NodeSubgraph<Self>
+        where
+            Self: StaticGraphViewOps;
+}
+
+impl LargestConnectedComponent for Graph {
+    fn largest_connected_component(&self) -> NodeSubgraph<Self>
+        where
+            Self: StaticGraphViewOps,
+    {
+        let connected_components_map = weakly_connected_components(self, usize::MAX, None).get_all_with_names();
```

Comment:
The Algorithm result has a group_by function which should allow you to easily get both the largest component and the ids of the nodes within it

Reply:
That sounds good. 👍